apisix/balancer.lua

```diff
@@ -308,15 +309,24 @@ function _M.run(route, ctx)
             return core.response.exit(502)
         end
 
+        plugin_funcs("balancer")
+
         local pass_host = ctx.pass_host
         if pass_host == "node" and balancer.recreate_request then
             local host = server.domain or server.host
             if host ~= ctx.var.upstream_host then
                 -- retried node has a different host
                 ctx.var.upstream_host = host
-                balancer.recreate_request()
+                header_changed = true
             end
         end
+
+    end
+
+    local _, run = plugin_funcs("balancer")
```

Comment:
We will return balancer twice?
Line 312 and line 326 ?

Reply:
Fixed in the new commit